Simulation/Models/ElectricPropellerModel_Generator.m

```diff
@@ -0,0 +1,64 @@
+function [Gen_ElectricPropellerModel] = ElectricPropellerModel_Generator(BladeDiameter, BladePitch, MotorKv, MaxCurrent, BatteryModel)
+  % This function generates a function delegate using the input parameters for
+  % an Electrically-powered Propeller Thrust Model. The model calculates the thrust force in X and Y directions
+  % and applies it to the derivatives vector as an acceleartion.
+  %
+  % Parameters:
+  %     BladeDiameter - Diameter of the propeller blade (inches)
+  %     BladePitch    - Pitch of the propeller blade (inches)
+  %     MotorKv       - Kv rating (velocity constant) of the motor (RPM/V)
+  %     MaxCurrent    - Maximum current consumed by the motor (A)
+  %     BatteryModel  - Battery model delegate used to calculate the current battery voltage supplied
+  %                     to the motor (Delegate that accepts battery capacity in Ah and returns voltage in V)
+  %
+  % Example:
+  %     PlaneBattery = BatteryModel_Generator(MaxVoltage, MinVoltage, Capacity);
+  %     ThrustModel  = ElectricPropellerModel_Generator(..., PlaneBattery);
+
+
+  Gen_ElectricPropellerModel = @ElectricPropellerModel;
+
+  function [Dervs] = ElectricPropellerModel(stateVector, controlsVector)
+    SV_Index = StateVector_Index;
+    CV_Index = ControlsVector_Index;
+    Dervs = zeros(11, 1);
+
+    throttle = controlsVector(CV_Index.Throttle);
+    voltage = BatteryModel(stateVector) * throttle;
+    airspeed = norm(stateVector(SV_Index.V));
+    RPM = MotorKv * voltage;
+
+    if stateVector(SV_Index.B_Cap) == 0

```

Comment:
The battery capacity currently doesn't have any implicit checks to see if it has passed 0 capacity thus you can have states within the Runge Kutta solver where you are passing through this check since a negative value inherently doesn't equal 0.

To fix this, we will need either additional code above this check to set a local battery capacity check to 0 before going into this logic, or switch the logic check to `<=0`;

Reply:
Similar thing goes here. 0 would be a valid battery capacity but anything below it wouldn't. If we do the centralized validation thing I mentioned above, the state vector should never contain an invalid value for the battery capacity or any other element.

Reply:
Same as the other validation comment, we will need validation to happen in the models to reflect proper behavior of the models locally.